src/python/pants/invalidation/cache_manager.py

```diff
@@ -172,20 +172,34 @@ def __init__(self, cache_manager, target, cache_key):
     super(VersionedTarget, self).__init__(cache_manager, [self])
     self.id = target.id
 
+  def _results_dir_prefix(self, root_dir):
+    """Return the prefix of the results directory path.
+
+    The prefix is qualified by the task version, and a stable symlink is
+    created to the currently active prefix, for convenience.  Note that this
+    stable symlink is distinct from the stable symlinks to specific results
+    directories under this prefix.
+    """
+    task_version = self._cache_manager.task_version
+    task_version_prefix = os.path.join(root_dir, sha1(task_version).hexdigest()[:12])
+    stable_prefix = os.path.join(root_dir, self._STABLE_DIR_NAME)
+    safe_mkdir(task_version_prefix)
+    safe_delete(stable_prefix)
```

Comment:
Have you tested these three statements for backwards compatibility with old workdirs? It looks like it should work, but symlinks can be tricky.